osu.Game/Rulesets/Difficulty/PerformanceBreakdownCalculator.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using osu.Framework.Extensions;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+
+namespace osu.Game.Rulesets.Difficulty
+{
+    public class PerformanceBreakdownCalculator
```

Comment:
Could make this a `Component`, at which point you could  use DI the same way as you would from a standard drawable, so long as the calculator is `Add()`ed to another composite drawable. Maybe not necesssary though.

Reply:
Probably not necessary. Feels weird to add a calculator to draw hierarchy just for DI.